Sources/Filters/General/ContourTriangulator/index.js

```diff
@@ -0,0 +1,2289 @@
+    let m = innerPoly.length;
+
+    for (let jj = 0; jj < m; jj++) {
+      // Semi-randomize the point order
+      let kk = (jj >> 1) + (jj & 1) * ((m + 1) >> 1);
+      let p = [];
+      points.getPoint(innerPoly[kk], p);
+
+      if (vtkPolygon.pointInPolygon(p, n, pp, bounds, normal)) {
+        let pointOnEdge = 0;
+        let q1 = [], q2 = [];
+        points.getPoint(outerPoly[n - 1], q1);
+
+        for (let ii = 0; ii < n; ii++) {
+          points.getPoint(outerPoly[ii], q2);
+          let t, dummy = [];
+          // This method returns distance squared
+          if (vtkLine.distanceToLine(p, q1, q2, t, dummy) < tol2) {
+            pointOnEdge = 1;
+            break;
+          }
+          q1[0] = q2[0];
+          q1[1] = q2[1];
+          q1[2] = q2[2];
+        }
+
+        if (!pointOnEdge) {
+          // Good result, point is in polygon
+          return 1;
+        }
+      }
+    }
+
+    // No matches found
+    return 0;
+  }
+
+  // ---------------------------------------------------
+  // Precompute values needed for the PolyInPoly check.
+  // The values that are returned are as follows:
+  // pp: an array of the polygon vertices
+  // bounds: the polygon bounds
+  // tol2: a tolerance value based on the size of the polygon
+  // (note: pp must be pre-allocated to the 3*outerPoly.length)
+
+  function vtkCCSPrepareForPolyInPoly(outerPoly, points, pp, bounds) {
+    const n = outerPoly.length;
+
+    if (n == 0) {
+      return 0.0; // to avoid false positive warning about uninitialized value
+    }
+
+    // Pull out the points
+    for (let k = 0; k < n; k++) {
+      // double* p = &pp[3 * k];
+      // points.getPoint(outerPoly[k], p);
+      pp[3*k] = outerPoly[k];
+      pp[3*k+1] = outerPoly[k+1];
+      pp[3*k+2] = outerPoly[k+2];
+    }
+
+    // Find the bounding box and tolerance for the polygon
+    return vtkCCSPolygonBounds(outerPoly, points, bounds) * (VTK_CCS_POLYGON_TOLERANCE * VTK_CCS_POLYGON_TOLERANCE);
+  }
+
+  // ---------------------------------------------------
+  // Check for polygons within polygons.  Group the polygons
+  // if they are within each other.  Reverse the sense of
+  // the interior "hole" polygons.  A hole within a hole
+  // will be reversed twice and will become its own group.
+
+  function vtkCCSMakeHoleyPolys(newPolys, points, polyGroups, polyEdges, originalEdges, normal, oriented) {
+    const numNewPolys = newPolys.length;
+    if (numNewPolys <= 1) {
+      return;
+    }
+
+    // Use bit arrays to keep track of inner polys
+    const polyReversed = new vtkCCSBitArray();
+    const innerPolys = new vtkCCSBitArray();
+
+    // GroupCount is an array only needed for unoriented polys
+    // TODO: Check
+    // size_t* groupCount = null;
+    // if (!oriented) {
+    //   groupCount = new size_t[numNewPolys];
+    //   std::fill(groupCount, groupCount + numNewPolys, 0);
+    // }
+    let groupCount;
+    if (!oriented) {
+      groupCount = [];
```

Comment:
`groupCount = new Int32Array(numNewPolys);` (that way you do not have to call `fill()`)